### DISCUSSION TITLE
[@types/webextension-polyfill] Error: Cannot find name 'Element'

### DISCUSSION THREAD

**Discussion by Eros1990** [DefinitelyTyped/DefinitelyTyped#60434] [Issues with a @types package] [answered]
In a project with this kind of typescript configuration:
```
"compilerOptions": {
        ....
        "lib": [
            "webworker",
        ],
       ....
    }
```
At building time an error is thrown:
`node_modules/@types/webextension-polyfill/namespaces/menus.d.ts:482:52 - error TS2304: Cannot find name 'Element'`.
Working on a MV3 web extension, a typescript configuration as the aforementioned is not uncommon.

[a comment by an automated account is not shown]

**Comment by Lusito** [comment, marked as answer]
Hi @Eros1990, Element is defined in the typescript dom libraries. You'll need to add "dom" and "dom.iterable" to compilerOptions.libs.

If you don't want those types in your setup (because you only work with webworkers, but have no UI), you can try these to shut up the error:
- Either add `"skipDefaultLibCheck": true,` to your compilerOptions
- Or possibly use a `declare global` statement to define the interface Element.

If you have an idea that could solve this differently (without breaking the other use-cases), feel free to create a PR: https://github.com/Lusito/webextension-polyfill-ts/issues

**Reply by Eros1990** [reply]
Thanks for your suggestions @Lusito.

I just created a minimal project with "WebWorker" as unique lib in 'compilerOptions.libs' and different errors are thrown.
Looking closely at the code of '@types/webextension-polyfill' there are multiple references to the dom elements and I think cannot be otherwise. 

I'll try to compile my project by suppressing these errors thrown by the typescript type check.

Sorry for wasting your time.
